### ISSUE TITLE
I trained my own dataset. But loss didn't decrease

### ISSUE THREAD

**Issue by tangdong1994** [amdegroot/ssd.pytorch#298]
My datasets has 5 classes. batch size = 8, lr=1e-5. others config just the same as orginal code.
**Here is my trained progress!**
![image](https://user-images.githubusercontent.com/37134249/53781431-ca3f1000-3f43-11e9-9a5d-025487790ae1.png)
![image](https://user-images.githubusercontent.com/37134249/53781448-d9be5900-3f43-11e9-858a-099ae6a1fac6.png)
![image](https://user-images.githubusercontent.com/37134249/53781465-e6db4800-3f43-11e9-9342-29ffd4abbe82.png)
![image](https://user-images.githubusercontent.com/37134249/53781478-f5c1fa80-3f43-11e9-805e-070d4328849f.png)
Is there a problem here?

**Comment by tangdong1994** [comment]
@amdegroot @ellisbrown @blackyang

**Comment by tangdong1994** [comment]
What is the reason why the loss function does not decrease but has an increasing trend?@amdegroot @ellisbrown @blackyang

**Comment by wendyltan** [comment]
I got the same problem bro.Anyone get a solution？

**Comment by tangdong1994** [comment]
> I got the same problem bro.Anyone get a solution？

I am trying to solve this problem by adjusting the learning rate and batch size. It seems to have some effect, but it is not completely solved. So maybe learning rate is too large

**Comment by tangdong1994** [comment]
> I got the same problem bro.Anyone get a solution？

hey bro. had solved it?I solved this problem by modifying code in box_utils.py.
`loc = encode(matches, priors, variances)
  loc_t[idx] = loc `
just Comment out first code，like following：
  ‘#loc = encode(matches, priors, variances)
   loc_t[idx] = matches

**Comment by toggle1995** [comment]
> > I got the same problem bro.Anyone get a solution？
> 
> hey bro. had solved it?I solved this problem by modifying code in box_utils.py.
> `loc = encode(matches, priors, variances) loc_t[idx] = loc `
> just Comment out first code，like following：
> ‘#loc = encode(matches, priors, variances)
> loc_t[idx] = matches

hey bro. I am using this code too. I only detect one class. Although the model can detect correctly,  I find the confidence probability outputted in test is not high, many targets only have the confidence of 0.3. Can you let me know your situation?

**Comment by tangdong1994** [comment]
> hey bro. I am using this code too. I only detect one class. Although the model can detect correctly, I find the confidence probability outputted in test is not high, many targets only have the confidence of 0.3. Can you let me know your situation?

OK,but I just use this code to train my model. my situation : detect five class. when i train my model the                 train loss increase not converge. so i found that code caused train loss increasing not converge

**Comment by YYZ-rose** [comment]
i meet the same problem, i try this modle by my data,but loss don't decrease. ok tanggong this name is chinese name. 这个损失是什么鬼！我到2万多轮损失还是2千多 @tangdong1994

**Comment by leiyaohui** [comment]
I got the same problem bro.Anyone get a solution？

**Comment by AlexanderSlav** [comment]
> > > I got the same problem bro.Anyone get a solution？
> > 
> > 
> > hey bro. had solved it?I solved this problem by modifying code in box_utils.py.
> > `loc = encode(matches, priors, variances) loc_t[idx] = loc `
> > just Comment out first code，like following：
> > ‘#loc = encode(matches, priors, variances)
> > loc_t[idx] = matches
> 
> hey bro. I am using this code too. I only detect one class. Although the model can detect correctly, I find the confidence probability outputted in test is not high, many targets only have the confidence of 0.3. Can you let me know your situation?
Hey, how can i contact with you, i have the same task, but i have some problems with training ssd on my dataset

**Comment by ahuxjz** [comment]
> > 嗨！哥们儿。我也在使用这段代码。我只检测一堂课。尽管模型可以正确检测，但我发现测试输出的置信度不高，许多目标的置信度仅为0.3。你能告诉我你的情况吗？
> 
> 好的，但是我只是使用此代码来训练我的模型。我的情况：检测五个班级。当我训练模型时，火车损失的增加不会收敛。所以我发现代码导致火车损失增加而无法收敛

Hello, has your loss problem been solved?  After 17,000 iterations, I still oscillate at 5,6,7 and occasionally jump above 10.

**Comment by JTzhuang** [comment]
There seems to be some problem with data preprocessing, The author did not normalize the data to (0,1) or (-1,1).

**Comment by qianchenghao** [comment]
> > I got the same problem bro.Anyone get a solution？
> 
> hey bro. had solved it?I solved this problem by modifying code in box_utils.py.
> `loc = encode(matches, priors, variances) loc_t[idx] = loc `
> just Comment out first code，like following：
> ‘#loc = encode(matches, priors, variances)
> loc_t[idx] = matches

Great work buddy.It works for me.But why?@tangdong1994

**Comment by Bobby2090** [comment]
> 
> 
> > > 嗨！哥们儿。我也在使用这段代码。我只检测一堂课。尽管模型可以正确检测，但我发现测试输出的置信度不高，许多目标的置信度仅为0.3。你能告诉我你的情况吗？
> > 
> > 
> > 好的，但是我只是使用此代码来训练我的模型。我的情况：检测五个班级。当我训练模型时，火车损失的增加不会收敛。所以我发现代码导致火车损失增加而无法收敛
> 
> Hello, has your loss problem been solved? After 17,000 iterations, I still oscillate at 5,6,7 and occasionally jump above 10.

嗨，你的问题解决了吗？我最近训练，也遇到同样的问题，求指导，谢谢

**Comment by Bobby2090** [comment]
> 
> 
> > > 嗨！哥们儿。我也在使用这段代码。我只检测一堂课。尽管模型可以正确检测，但我发现测试输出的置信度不高，许多目标的置信度仅为0.3。你能告诉我你的情况吗？
> > 
> > 
> > 好的，但是我只是使用此代码来训练我的模型。我的情况：检测五个班级。当我训练模型时，火车损失的增加不会收敛。所以我发现代码导致火车损失增加而无法收敛
> 
> Hello, has your loss problem been solved? After 17,000 iterations, I still oscillate at 5,6,7 and occasionally jump above 10.

嗨，你的问题解决了吗？我最近训练，也遇到同样的问题，求指导，谢谢
